### PR TITLE
update clickhouse and postgres images in Makefile

### PR DESCRIPTION
**Pull request by ruslandoga**
### Changes

This PR updates Makefile's `postgres` version [to the one used in `plausible/hosting` (version 14)](https://github.com/plausible/hosting/blob/a6fe8d13df3566f7bb3625a1a524f5498d56693d/docker-compose.yml#L9) and `clickhouse` -- to the version [that replaced `hyperscan` with `vectorscan` for better arm64 support (version 22.9)](https://github.com/ClickHouse/ClickHouse/pull/38171)

Relevant:
- https://github.com/plausible/analytics/discussions/2292
- https://github.com/plausible/hosting/pull/57
- https://github.com/plausible/analytics/discussions/2342

### Tests
- [ ] Automated tests have been added
- [x] This PR does not require tests

### Changelog
- [ ] Entry has been added to changelog
- [x] This PR does not make a user-facing change

### Documentation
- [ ] [Docs](https://github.com/plausible/docs) have been updated
- [x] This change does not need a documentation update

### Dark mode
- [ ] The UI has been tested both in dark and light mode
- [x] This PR does not change the UI

### CODE REVIEW

**Comment by aerosol** [comment]
What'd be the best way to keep the prod parity purely for development? 
Self hosting is bleeding edge I see but for regular dev we should keep the same versions we run on prod. Would it make sense to maintain separate, additional make targets for that?

**Comment by ruslandoga** [comment]
@aerosol additional targets for prod or bleeding edge versions?

**Comment by aerosol** [comment]
> I think the former -- make clickhouse-prod && make postgres-prod -- hurts less since only the core team would need to use them. 

I think that's fine. My only concern is we should keep track of (and provide access to) the version that's currently live for both dev and test/CI.

**Comment by aerosol** [comment]
> and test/CI

speaking of.. looks like we're running the CI against yet something else https://github.com/plausible/analytics/blob/json-decode-with-captcha-test/.github/workflows/elixir.yml#L32 - not sure if you like to address it in this PR?

File: Makefile [hunk 1]
```diff
@@ -1,27 +1,40 @@
...
 
-postgres:
-	docker run --detach -e POSTGRES_PASSWORD="postgres" -p 5432:5432 --volume=plausible_db:/var/lib/postgresql/data --name plausible_db postgres:12
+PG_FLAGS ?= --detach -e POSTGRES_PASSWORD="postgres" -p 5432:5432 --name plausible_db
+
+postgres: ## Run a container with a recent version of postgres
+	docker run $(PG_FLAGS) --volume=plausible_db:/var/lib/postgresql/data postgres:14-alpine
+
+postgres-prod: ## Run a container with the same version of postgres as the one in prod
+	docker run $(PG_FLAGS) --volume=plausible_db_prod:/var/lib/postgresql/data postgres:12
```
Comment (ruslandoga) [inline Makefile:37]: same for `postgres-prod`

File: Makefile [hunk 2]
```diff
@@ -1,27 +1,40 @@
-.PHONY: install server clickhouse clickhouse-arm clickhouse-stop postgres postgres-stop
+.PHONY: help install server clickhouse clickhouse-arm clickhouse-stop postgres postgres-stop
 
-install:
+help:
```
Comment (ruslandoga) [inline Makefile:3]:
```console
$ make help
clickhouse-ci                  Run a container with the same version of clickhouse as the one in .github/workflows/elixir.yml
clickhouse-prod                Run a container with the same version of clickhouse as the one in prod
clickhouse-stop                Stop and remove the clickhouse container
clickhouse                     Run a container with a recent version of clickhouse
install                        Run the initial setup
postgres-prod                  Run a container with the same version of postgres as the one in prod
postgres-stop                  Stop and remove the postgres container
postgres                       Run a container with a recent version of postgres
server                         Start the web server
```

Comment (aerosol) [reply Makefile:3]: clever :heart_eyes:

File: Makefile [hunk 3]
```diff
@@ -1,27 +1,40 @@
...
 
-clickhouse:
-	docker run --detach -p 8123:8123 --ulimit nofile=262144:262144 --volume=$$PWD/.clickhouse_db_vol:/var/lib/clickhouse --name plausible_clickhouse clickhouse/clickhouse-server:21.11.3.6
+CH_FLAGS ?= --detach -p 8123:8123 --ulimit nofile=262144:262144 --name plausible_clickhouse
+
+clickhouse: ## Run a container with a recent version of clickhouse
+	docker run $(CH_FLAGS) --volume=$$PWD/.clickhouse_db_vol:/var/lib/clickhouse clickhouse/clickhouse-server:22.9-alpine
+
+clickhouse-prod: ## Run a container with the same version of clickhouse as the one in prod
+	docker run $(CH_FLAGS) --volume=$$PWD/.clickhouse_db_vol_prod:/var/lib/clickhouse clickhouse/clickhouse-server:21.11.3.6
```
[the remark this answers is not in the document]
Comment (aerosol) [reply Makefile:23]: I think this is great and we can re-iterate from here. cc @dch

**Review by aerosol — APPROVED** [review]
Very nice :clap:

File: Makefile (hunk 3 above)
[the remark this answers is not in the document]
Comment (aerosol) [reply Makefile:23]: We'll have to change the image that runs on CI too, putting it on my TODO

File: Makefile [hunk 4]
```diff
@@ -1,27 +1,40 @@
-.PHONY: install server clickhouse clickhouse-arm clickhouse-stop postgres postgres-stop
+.PHONY: help install server clickhouse clickhouse-arm clickhouse-stop postgres postgres-stop
```
[the remark this answers is not in the document]
Comment (ruslandoga) [reply Makefile:1]: oops, right!

File: Makefile (hunk 2 above)
Comment (ruslandoga) [reply Makefile:3]: copied it from https://github.com/bitwalker/distillery-aws-example/blob/f859eb101c55f8332c40db8ed64de55262416fad/Makefile#L7-L9

**Pull request merged**